src/Build/Logging/EvaluationLocationIdAgnosticComparer.cs

```diff
@@ -36,13 +36,13 @@ public bool Equals(EvaluationLocation x, EvaluationLocation y)
         public int GetHashCode(EvaluationLocation obj)
         {
             var hashCode = 1198539463;
-            hashCode = hashCode * -1521134295 + obj.EvaluationPass.GetHashCode();
+            hashCode = hashCode * -1521134295 + EqualityComparer<EvaluationPass>.Default.GetHashCode(obj.EvaluationPass);
             hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(obj.EvaluationPassDescription);
             hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(obj.File);
             hashCode = hashCode * -1521134295 + EqualityComparer<int?>.Default.GetHashCode(obj.Line);
             hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(obj.ElementName);
             hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(obj.ElementDescription);
-            hashCode = hashCode * -1521134295 + obj.Kind.GetHashCode();
+            hashCode = hashCode * -1521134295 + EqualityComparer<EvaluationLocationKind>.Default.GetHashCode(obj.Kind);
```

Comment:
I wasn't familiar with this idiom, and when I tried putting it into [sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgAwAEWeALANzFYDMlO5AwuQN7Hm+XNZq5ALIAKAJRcefGQDtyAS3IBecrIB0AMwVwANgBM89IjJlUAnKIBEACwVXxx03wWyY5G9Bt4Vi9QHE4GAAJLzYAe304CSdnSjxLTyhvR2lnV3ckm1ZVAFEARwBXCF0FGABPCLAABwgERAAeWQA+dQAROE0IQt0YAKDQ5Iio0QVUkziLUSyccZkAXzTyaoQFADcIeHI4WUKwNSkJ5209QyRyE4McJcWieaA=), it seemed to do the same thing. What did I do wrong?

Reply:
`constrained` can box: https://sharplab.io/#v2:EYLgdgpgLgZgHgGiiAlgGwD4AEAMACLARgBYBuAWACgIwBXAWzwFE8BvAXyqIDYCAmPAGE2VPGIIBmAoV4owUPAFlCAChYQAlHgC8APjwQAdAHFoACQCGAZwAWggPYATCCo0VK4ydNnylfNQZaeswAjrQWaChQAJ4O9AAOFgBOEEkAPEy6hgAiEDAWtGhQJubWdk4umu7sQA

Reply:
One super hacky way to quickly check if a piece of code allocates is simply running it in a tight `while (true)` loop in a trivial console app and checking the VS diagnostics window.

Seeing small yellow markers appear regularly means that GC is collecting something and that something has to be allocations performed in the loop.

![image](https://user-images.githubusercontent.com/12206368/89864253-eec50000-dbab-11ea-8173-8b42ecb39436.png)